### PR TITLE
Ak migration stats

### PR DESCRIPTION
**Pull request by aravindhank11**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,4 +9,6 @@
 *output_nsys.sqlite
 model-formulation/catboost_info/*
 print_outs.txt
+src/*_pb2*.py
 *__pycache__*
+*.zip
```
Comment (psinha25) [inline .gitignore:14]: What zip file is created?

File: src/scheduler.py [hunk 2]
```diff
@@ -197,10 +197,10 @@ def UpdateClusterState(self, request, context):
                 updated_state['mps'].num_available -= 1
         # Log completed runs to retrieve stats post processing from kafka
         elif request.reason == 'end mix':
-            echo_cmd = f"echo {request.run_id} >> /tmp/tiebreaker-completed-mixes.txt"
+            echo_cmd = f'[ -p "/tmp/{request.run_id}.client-wait" ] && echo "1" > /tmp/{request.run_id}.client-wait'
```
Comment (psinha25) [inline src/scheduler.py:200]: Why this change?

**Review by psinha25 — COMMENTED** [review]
Looks good, two quick questions.

File: src/scheduler.py (hunk 2 above)
Comment (aravindhank11) [inline src/scheduler.py:200]: For each client, we will write into a fifo file that the client creates. If there is no file, then scheduler does nothing. This both the logics are simple.

File: .gitignore (hunk 1 above)
Comment (aravindhank11) [reply .gitignore:14]: While install grpc it downloads some zip

File: src/scheduler.py (hunk 2 above)
Comment (aravindhank11) [reply src/scheduler.py:200]: This is the mkfifo change I had proposed. But in earlier proposal there was a blocking nature as I suggested the scheduler to create the fifo file. Now in this change, the client creates the fifo file. If the file exists the scheduler echoes there by there is no blocking nature.

File: .gitignore (hunk 1 above)
Comment (aravindhank11) [reply .gitignore:14]: from https://github.com/aravindhank11/llama2-expr/pull/14/files#diff-52ed8b9ee32a55b12592adf96a1f1fb0eb7efc80831eedb3132677e53497aff7R39

This was moved from a separate install file to the common one

File: src/scheduler.py (hunk 2 above)
Comment (psinha25) [reply src/scheduler.py:200]: Ah okay, makes sense.

Comment (psinha25) [reply src/scheduler.py:200]: Great, thanks for this change.

File: .gitignore (hunk 1 above)
Comment (psinha25) [reply .gitignore:14]: Ah yes, thanks
